src/main/java/io/cdap/plugin/gcp/gcs/actions/GCSArgumentSetter.java

```diff
@@ -106,12 +106,12 @@ public void run(ActionContext context) throws Exception {
   }
 
   private static Storage getStorage(GCSArgumentSetterConfig config) throws IOException {
-    return StorageOptions.newBuilder()
-      .setProjectId(config.getProject())
-      .setCredentials(GCPUtils.loadServiceAccountCredentials(config.getServiceAccount(),
-                                                             config.isServiceAccountFilePath()))
-      .build()
-      .getService();
+    String serviceAccount = config.getServiceAccount();
+    StorageOptions.Builder builder = StorageOptions.newBuilder().setProjectId(config.getProject());
+    if (serviceAccount != null) {
+      builder.setCredentials(GCPUtils.loadServiceAccountCredentials(serviceAccount, config.isServiceAccountFilePath()));
```

Comment:
What would be credentials when service account is null?

Reply:
Also `config.isServiceAccountFilePath()` can be null. Wouldnt `GCPUtils#loadServiceAccountCredentials` method throw a null pointer exception when NPE? Has this change been tested with that case?

Reply:
when service account is null credentials are not set and the library loads default credentials as in other plugins. Regarding ``config.isServiceAccountFilePath()`` being null - this condition cannot be fullfilled due to check in line 111, if we have serviceAccount that means that serviceAccountType is not null.

Reply:
I see